bug(is_a:) typo causes schemaloader recursion error hard to debug

**Describe the bug**
A clear and concise description of what the bug is.

**To Reproduce**
Steps to reproduce the behavior:

1. Define a Slot  'AAAA'
2. Define a Slot 'BBBB' with property `is_a: AAAAA`   (intentional typo) 

Run `linkml/utils/schemaloader.py` against the linkml model to produce error:

```
 self.merge_slot(self.schema.slots[slot.is_a], merged_slots)
  [Previous line repeated 982 more times]
  File "/home/runner/.local/share/virtualenvs/csolink-model-GmhkEpOf/lib/python3.7/site-packages/linkml/utils/schemaloader.py", line 528, in merge_slot
    if slot.name not in merged_slots:
RecursionError: maximum recursion depth exceeded in comparison
make: *** [Makefile:53: csolink/model.py] Error 1
```

The typo (non existing slot) is probably triggered by this block:

```
528         if slot.name not in merged_slots:
529             if slot.is_a:
530                 if slot.is_a in self.schema.slots:
531                     self.merge_slot(self.schema.slots[slot.is_a], merged_slots)
532                     merge_slots(slot, self.schema.slots[slot.is_a])
```


**Expected behavior**

I had to debug the model to find the typo.  Maybe the following pseudo code would help debug this:

```
        try:
        
            <recursive call>      

        catch RecursionError:
              print("slot name %s is not defined" % slot.name)
              exit
```

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Desktop (if applicable, please complete the following information):**
 - OS: [e.g. iOS]
 - Browser [e.g. chrome, safari]
 - Version [e.g. 22]

**Additional context**
Add any other context about the problem here.

--- a/linkml/utils/schemaloader.py
+++ b/linkml/utils/schemaloader.py
@@ -527,11 +527,15 @@ class SchemaLoader:
         """
         if slot.name not in merged_slots:
             if slot.is_a:
-                if slot.is_a in self.schema.slots:
-                    self.merge_slot(self.schema.slots[slot.is_a], merged_slots)
-                    merge_slots(slot, self.schema.slots[slot.is_a])
-                else:
-                    self.raise_value_error(f'Slot: "{slot.name}" - unknown is_a reference: {slot.is_a}', slot.is_a)
+                try:
+                    if slot.is_a in self.schema.slots:
+                        self.merge_slot(self.schema.slots[slot.is_a], merged_slots)
+                        merge_slots(slot, self.schema.slots[slot.is_a])
+                    else:
+                        self.raise_value_error(f'Slot: "{slot.name}" - unknown is_a reference: {slot.is_a}', slot.is_a)
+                except RecursionError:
+                        self.raise_value_error(f'Slot: "{slot.name}" - recursive is_a reference: {slot.is_a}', slot.is_a)
+
             for mixin in slot.mixins:
                 if mixin in self.schema.slots:
                     self.merge_slot(self.schema.slots[mixin], merged_slots)

